core/common/src/InstantParser.kt

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2019-2021 JetBrains s.r.o.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.datetime
+
+import kotlin.math.min
+import kotlin.math.pow
+
+internal fun parseInstantCommon(string: String): Instant = parseIsoString(string)
```

Comment:
The parseInstantCommon facade may try multiple formats before giving up such as #83 if that gets implemented.